core/sql/common/Collections.h

```diff
@@ -2389,10 +2389,15 @@ template <class T> class NASimpleArray : private NAArray<T>
   
   // operator s new and delete get inherited from private base class,
   // make them publicly available by the following inline functions
-  static inline void * operator new(size_t size, CollHeap* h=0)
+  static inline void * operator new(size_t size, CollHeap* h)
                                 { return NAArray<T>::operator new(size, h); }
   static inline void operator delete(void *buffer)
                                   { NAArray<T>::operator delete(buffer); }
+
+  inline void setHeap(CollHeap *heap)
```

Comment:
May not be good idea to do setHeap after it is constructed. This can possibly result in different heaps to be used for different elements and cause core dumps